src/Lean/Elab/PreDefinition/PartialFixpoint/Induction.lean

```diff
@@ -57,25 +57,28 @@ I.e `ImplicationOrder.instPartialOrder.rel P Q` becomes
 In the premise of the Park induction principle (`lfp_le_of_le_monotone`) we use a monotone map defining the predicate in the eta expanded form. In such a case, besides desugaring the predicate, we need to perform a weak head reduction.
 The optional parameter `reduceConclusion` (false by default) indicates whether we need to perform this reduction.
 -/
-def desugarOrder (predType : Expr) (body : Expr) (reduceConclusion : Bool := false) (fixpointType : Option Bool) : MetaM Expr := do
-  if fixpointType.isNone then
-    throwError "Trying to apply lattice induction to a non-lattice fixpoint. Please report this issue."
-  if body.isAppOfArity ``PartialOrder.rel 4 then
+def desugarOrder (predType : Expr) (body : Expr) (fixpointType : PartialFixpointType) (reduceConclusion : Bool := false) : MetaM Expr := do
```

Comment:
Thanks for the docstring! Now I can suggest a better name, since “desugaring” is usually related to syntax. Maybe `unfoldPredRel`, since it unfolds the relation on predicates?